### PR TITLE
chore(storage-browser): set up e2e tests for storage browser

### PR DESCRIPTION
**Pull request by timngyn**
#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/e2e/features/ui/components/storage/storage-browser/render-storage-browser.feature [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    When I type my "email" with status "CONFIRMED"
+    Then I type my password
+    Then I click the "Sign in" button
+    Then I see the "Sign Out" button
+    Then I see "Home"
+    Then I see the "s3://fileuploaderbucket121216-dev/public/*" button
+    Then I click the "s3://fileuploaderbucket121216-dev/public/*" button
+    Then I see "test.txt"
+    Then I click the "Download public/test.txt" button
+    Then I verify the downloaded file "test.txt" has the contents "file contents"
```
Comment (timngyn) [inline packages/e2e/features/ui/components/storage/storage-browser/render-storage-browser.feature:26]: For some reason, this works when I do `yarn e2e dev` and then run it manually through the UI, but doesn't work when I'm just using the CLI. When running it through CLI,  I'm able to see the file get downloaded but it's not able to read it I'm guessing?

**Review by calebpollman — COMMENTED** [review]
Let's slow down with adding more tests until we have verified the approach for the ones already added

File: packages/e2e/features/ui/components/storage/storage-browser/error-boundary.feature [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    When I click the "Actions" button
+    Then I see the "Create Folder" menuitem
+    When I click the "Create Folder" menuitem
+    Then I see "Enter folder name:"
+    Then I see the "Create Folder" button
+    Then the "Create Folder" button is disabled 
+    When I type a new "Enter folder name:" with value "Blackberry"
+    Then I see the "Create Folder" button
+    When I click the "Create Folder" button
+    Then I see "no identity-based policy allows the s3:PutObject action"
```
Comment (jordanvn) [inline packages/e2e/features/ui/components/storage/storage-browser/error-boundary.feature:33]: I was not sure what the best way of identifying the error boundary would be. Here, I am ensuring that the text within the error appears on the page. I'm open to suggestions on a more effective method of testing this error boundary would be

File: packages/e2e/features/ui/components/storage/storage-browser/error-boundary.feature
Comment (jordanvn) [inline packages/e2e/features/ui/components/storage/storage-browser/error-boundary.feature:1]: These tests follow the same general actions as those used in the `insufficient-permissions` tests. Would it make more sense if they were combined into one set of error state tests? Or is it better to keep both targets distinct?

**Comment by jordanvn** [comment]
This will be posted for an initial review before carrying on with additional tests. For clarity, the additional tests to be added are:

* Upload:
  * Authed:
    * Able to upload a file
    * Able to upload multiple files
    * Able to upload a folder
    * Able to add files to the tasks table before upload
    * Able to add a folder to the tasks table before upload
    * Able to remove files from tasks table before upload
    * Able to exit the upload action
  * Unauthed: 
    * Unable to upload a file without WRITE permissions
    * Unable to upload a folder without WRITE permissions
* Create folder:
  * An error message is shown if a network error is encountered
* Misc.:
  * A loading spinner should be rendered before the table loads
    * This should occur at the home page, folder locations, and after refreshing

File: packages/e2e/features/ui/components/storage/storage-browser/insufficient-permissions.feature [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    When I click the "Actions" button
+    Then I see the "Create Folder" menuitem
+    When I click the "Create Folder" menuitem
+    Then I see "Enter folder name:"
+    Then I see the "Create Folder" button
+    Then the "Create Folder" button is disabled 
+    When I type a new "Enter folder name:" with value "Blackberry"
+    Then I see the "Create Folder" button
+    When I click the "Create Folder" button
+    Then I do not see "Folder created."
```
Comment (reesscot) [inline packages/e2e/features/ui/components/storage/storage-browser/insufficient-permissions.feature:32]: Missing newline character

File: packages/e2e/features/ui/components/storage/storage-browser/navigate-locations.feature [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    Then I click the "Sign in" button
+    When I click the button containing "public"
+    Then I see the "Blueberry/" button
+    When I click the "Blueberry/" button
+    Then I see the "Acai/" button
+    When I click the "Acai/" button
+    Then I see "Acai/"
+    Then I see the "Blueberry" button
+    When I click the "Blueberry" button
+    Then I see "Blueberry/"
```
Comment (reesscot) [inline packages/e2e/features/ui/components/storage/storage-browser/navigate-locations.feature:44]: Missing newline character

Comment (jordanvn) [reply packages/e2e/features/ui/components/storage/storage-browser/navigate-locations.feature]: Got it!

File: packages/e2e/features/ui/components/storage/storage-browser/insufficient-permissions.feature (hunk 3 above)
Comment (jordanvn) [reply packages/e2e/features/ui/components/storage/storage-browser/insufficient-permissions.feature]: Got it!

File: packages/e2e/features/ui/components/storage/storage-browser/error-boundary.feature
Comment (reesscot) [reply packages/e2e/features/ui/components/storage/storage-browser/error-boundary.feature:1]: I believe this test is supposed to test that throwing an unexpected error in the Component code should be caught in the React error boundary, not displayed as a simple message.

File: packages/e2e/features/ui/components/storage/storage-browser/insufficient-permissions.feature [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    Given I'm running the example "ui/components/storage/storage-browser/default-auth"
+
+  @react
+  Scenario: Attempt to access forbidden folder
+    When I type my "email" with status "CONFIRMED"
+    Then I type my password
+    Then I see the "Sign in" button
+    When I click the "Sign in" button
+    When I click the button containing "forbidden"
+    Then I don't see "amplify-table"
```
Comment (reesscot) [inline packages/e2e/features/ui/components/storage/storage-browser/insufficient-permissions.feature:13]: This seems pretty implementation specific. Can we check for a forbidden error or something?

**Review by dbanksdesign — APPROVED** [review]

**Review by hbuchel — APPROVED** [review]

File: packages/e2e/features/ui/components/storage/storage-browser/insufficient-permissions.feature (hunk 5 above)
Comment (jordanvn) [reply packages/e2e/features/ui/components/storage/storage-browser/insufficient-permissions.feature]: Updated to watch for a 403
